### ISSUE TITLE
Go to line command allows non-digit characters resulting in error message

### ISSUE THREAD

**Issue by sbauer322** [LightTable/LightTable#2308]
When using the `Editor: Go to line` command, it is possible to enter characters other than digits and press enter to jump to the line. Doing so results in an error message in Light Table's console. See below:

![go-to-line-non-digit-text](https://cloud.githubusercontent.com/assets/2945386/21575593/90dd5848-cedf-11e6-937e-d8debd02a880.png)

We should handle this situation better. Perhaps either prevent non-digit text from being accepted in the first place, or send along an alert/warning/error explaining that the text was invalid.

**Comment by bruno-oliveira** [comment]
Maybe when the other issue is finally solved and the PR is accepted, I can start working on this? Or is this not suited for beginners?

**Comment by sbauer322** [comment]
I'm not certain if it would be suitable for beginners, but then again it would largely come down to how we want to address the problem. 

You are certainly welcome to give it a try - our [Gitter](https://gitter.im/LightTable/LightTable) is a good place for any questions you may come across. I think the second solution mentioned below might be suitable for beginners.

It seems the problem lies in the `goto-line` (soon to be `go-to-line`) command incorrectly determining if there is a non-number input. 

In my opinion, it should be that we either:

- Prevent the user from submitting non-number inputs
- Or, and this might be best for now, log an error to the console stating that they the go to line input must be a number

If you or anyone else have another solution, feel free to give it a try... but there is no need to go overboard with something elaborate! :smiley:

**Comment by bruno-oliveira** [comment]
I am still learning clojure as well as clojurescript (following the 4clojure website at the moment), but, thinking purely in terms of general programming, shouldn't we simply add a regex verification that the entered input needs to match? Namely, that the number needs to be between 1 and MAX_LINES where MAX_LINES would be the total number of lines of the given file?

For this, I might have to setup the dev version locally and now that Im back working it might be more complicated for me, but, Ill try to keep up :)

**Comment by sbauer322** [comment]
That sounds about right in terms of logic, but the main concern I would have, if we went with regex validation, is how do we make the user aware their input is invalid?

There already exists some validation logic to check if it is a number, but I think there is a situation in a predicate where it evaluates to true if you enter non-numeric text (from what I remember after briefly looking at it a few days ago). Might stem from using `js/parseInt`.

Regarding keeping the number between 1 and MAX_LINES, I'm fine with the current behavior of appropriately going to the first or last line of the file when exceeding the actual range of acceptable values. I imagine there can be cases where the user doesn't know the exact number of lines.

**Comment by bruno-oliveira** [comment]
For this, I might have to setup a local dev version to be able to test te changes I'd be doing... I will try to do it over the weekend to start looking into it...
However, thinking about it, the regex validation can only address part of the problem, right?
One thing is to make sure that invalid input is caught/detected (which can be done via regex) and other thing, that needs to be done as a consequence of invalid/valid input is to either, go to the assigned line/beginning of the document if input is a number or sending an error message as user feedback is input is invalid.

I will setup a local development version over the weekend, and I'll get back to you once I have explored this issue a bit further.

**Comment by Mouvedia** [comment]
```
entered += char_entered.replace(/[^0-9]/, '');
entered = entered.replace(/^0+/, '');
```

tl;dr: you don't register visually anything but digits or you use an input of type number.
